modules/storage/minio.go

```diff
@@ -90,6 +95,19 @@ func NewMinioStorage(ctx context.Context, cfg *setting.Storage) (ObjectStorage,
 		return nil, convertMinioErr(err)
 	}
 
+	// Check if the connection works
+	err = getBucketVersioning(ctx, minioClient, config.Bucket)
+	if err != nil {
+		errResp, ok := err.(minio.ErrorResponse)
+		if !ok {
+			return nil, err
+		}
+		if errResp.StatusCode == http.StatusBadRequest {
```

Comment:
Why not handle other status codes, such as StatusNotFound, StatusForbidden, StatusConflict and StatusPreconditionFailed?

Reply:
It is explained in the PR description.